x-pack/plugins/apm/server/routes/alerts/rule_types/anomaly/register_anomaly_rule_type.ts

```diff
@@ -161,8 +173,14 @@ export function registerAnomalyRuleType({
                       },
                     },
                   },
-                  ...termQuery('partition_field_value', ruleParams.serviceName),
-                  ...termQuery('by_field_value', ruleParams.transactionType),
+                  ...termQuery(
+                    'partition_field_value',
+                    ruleParams.serviceName || null
```

Comment:
Why is this needed? Is `ruleParams.serviceName`  not `null` when all is selected?

Reply:
in development, `ruleParams.serviceName` returned an empty string, which the `termQuery` function doesn't omit. This was a quick fix, but it should probably be fixed either within `termQuery` or `ruleParams.serviceName`

Reply:
the fix is in `x-pack/plugins/apm/public/components/alerting/utils/fields.tsx`, to make sure the `onChange` handler is passed `undefined` when the field has value `''`.